src/dotnet/Chat.UI.Blazor/Components/ChatList/ChatList.razor

```diff
@@ -1,28 +1,67 @@
 @using ActualChat.Users
 @namespace ActualChat.Chat.UI.Blazor.Components
+@inherits ComputedStateComponent<ChatList.Model>
 @{
-    if (Chats == null || Chats.Count == 0)
-        return;
+    var m = State.Value;
+    var chatCount = m.ChatCount ?? 20;
+    var isLoaded = m != Model.Loading;
+    _lastRendered = m;
 }
 
-<div class="@GetCss(Kind) chat-list">
-    @foreach (var chat in Chats) {
-        <ChatListItem @key="@chat" ChatInfo="@chat" ListKind="@Kind"/>
-    }
-</div>
-@if (Kind == ChatListKind.Pinned) {
-    <div class="mx-4 mb-2 border-b border-bg-04"></div>
+@if (isLoaded) {
+    <div class="chat-list chat-list-all">
+        @for (var i = 0; i < chatCount; i++) {
+            var position = i;
+
+            <ChatListItem @key="@position" Position="@position" ListKind="@ChatListKind.All" />
```

Comment:
why is index a key?

Reply:
Coz the idea is to make sure that only moved / updated items are re-rendered.

Reply:
Technically key isn't needed here, btw. And maybe it's even better to do it w/o a key, coz having key = chatCount boxings per render.

Reply:
I.e. w/o a key Blazor will simply map these components by index.

Reply:
I will try to get rid of it